airflow/contrib/hooks/google_discovery_api_hook.py

```diff
@@ -16,132 +16,14 @@
-            )
-        return self._conn
-
-    def query(self, endpoint, data, paginate=False, num_retries=0):
-        """
-        Creates a dynamic API call to any Google API registered in Google's API Client Library
-        and queries it.
-
-        :param endpoint: The client libraries path to the api call's executing method.
-            For example: 'analyticsreporting.reports.batchGet'
-
-            .. seealso:: https://developers.google.com/apis-explorer
-                for more information on what methods are available.
-        :type endpoint: str
-        :param data: The data (endpoint params) needed for the specific request to given endpoint.
-        :type data: dict
-        :param paginate: If set to True, it will collect all pages of data.
-        :type paginate: bool
-        :param num_retries: Define the number of retries for the requests being made if it fails.
-        :type num_retries: int
-        :return: the API response from the passed endpoint.
-        :rtype: dict
-        """
-        google_api_conn_client = self.get_conn()
-
-        api_response = self._call_api_request(google_api_conn_client, endpoint, data, paginate, num_retries)
-        return api_response
-
-    def _call_api_request(self, google_api_conn_client, endpoint, data, paginate, num_retries):
-        api_endpoint_parts = endpoint.split('.')
-
-        google_api_endpoint_instance = self._build_api_request(
-            google_api_conn_client,
-            api_sub_functions=api_endpoint_parts[1:],
-            api_endpoint_params=data
-        )
-
-        if paginate:
-            return self._paginate_api(
-                google_api_endpoint_instance, google_api_conn_client, api_endpoint_parts, num_retries
-            )
-
-        return google_api_endpoint_instance.execute(num_retries=num_retries)
-
-    def _build_api_request(self, google_api_conn_client, api_sub_functions, api_endpoint_params):
-        for sub_function in api_sub_functions:
-            google_api_conn_client = getattr(google_api_conn_client, sub_function)
-            if sub_function != api_sub_functions[-1]:
-                google_api_conn_client = google_api_conn_client()
-            else:
-                google_api_conn_client = google_api_conn_client(**api_endpoint_params)
-
-        return google_api_conn_client
-
-    def _paginate_api(
-        self, google_api_endpoint_instance, google_api_conn_client, api_endpoint_parts, num_retries
-    ):
-        api_responses = []
-
-        while google_api_endpoint_instance:
-            api_response = google_api_endpoint_instance.execute(num_retries=num_retries)
-            api_responses.append(api_response)
-
-            google_api_endpoint_instance = self._build_next_api_request(
-                google_api_conn_client, api_endpoint_parts[1:], google_api_endpoint_instance, api_response
-            )
+"""This module is deprecated. Please use `airflow.gcp.hooks.discovery_api`."""
```

Comment:
Actually, this Hook hasn't been released yet. So I think we can just move it to core without a note.

Reply:
Same for the `GoogleApiToS3Transfer`

Reply:
I agree. Can you also look at other operators and hooks?

Reply:
Good point